File: tracker/src/test/java/testhelpers/MatomoTestApplication.java
package testhelpers;


import org.matomo.sdk.TrackerBuilder;
import org.matomo.sdk.extra.MatomoApplication;
import org.robolectric.TestLifecycleApplication;
import org.robolectric.shadows.ShadowLog;

import java.lang.reflect.Method;

public class MatomoTestApplication extends MatomoApplication implements TestLifecycleApplication {

    @Override
    public void onCreate() {
        ShadowLog.stream = System.out;
        super.onCreate();
    }

    @Override
    public void beforeTest(Method method) {

    }

    @Override
    public void prepareTest(Object test) {
    }

    @Override
    public void afterTest(Method method) {

    }

    @Override
    public String getPackageName() {
        return "11";
    }


    @Override
    public TrackerBuilder onCreateTrackerConfig() {
        return TrackerBuilder.createDefault("http://example.com", 1);
    }
}

File: tracker/src/test/java/testhelpers/MatomoTestApplication.kt
package testhelpers

import org.matomo.sdk.TrackerBuilder
import org.matomo.sdk.extra.MatomoApplication
import org.robolectric.TestLifecycleApplication
import org.robolectric.shadows.ShadowLog
import java.lang.reflect.Method


class MatomoTestApplication : MatomoApplication(), TestLifecycleApplication {
    override fun onCreate() {
        ShadowLog.stream = System.out
        super.onCreate()
    }

    override fun beforeTest(method: Method) {
    }

    override fun prepareTest(test: Any) {
    }

    override fun afterTest(method: Method) {
    }

    override fun getPackageName(): String {
        return "11"
    }


    override fun onCreateTrackerConfig(): TrackerBuilder {
        return TrackerBuilder.createDefault("http://example.com", 1)
    }
}
